calkit/cli/main.py

```diff
@@ -1145,8 +1179,17 @@ def run(
         if save_message is None:
             save_message = "Run pipeline"
         if not quiet:
-            typer.echo("Saving the project after successful run")
+            typer.echo("Saving the whole project after successful run")
         save(save_all=True, message=save_message)
+    if save_successes_after_run:
```

Comment:
I think this will need to be moved up into the `if failed` block, and then raise the error. Also, if `save_log` is `False`, we should probably use either a temp file or an `io.StringIO` object so we don't inadvertently write logs that get committed to the repo.